tapestry.js

```diff
@@ -1081,8 +1172,8 @@ function setDatasetProgress(progressObj) {
 function setNodeTypes(rootId) {
 
     root = rootId;
-    var children = getChildren(root),
-        grandchildren = getGrandchildren(children);
+    var children = getChildrenRec(root,locn-1),
```

Comment:
I see that you're passing in `locn-1` a lot into your function. To make it more brief, you could change your function to have the second parameter as optional so that you don't have to pass a second parameter if it's `locn-1`. In your `getChildrenRec` function, you could add a line like
```
if (typeof depth === 'undefined') {
    depth = locn-1;
}
```
so this will act as a default for that second parameter.